pkg/registry/batch/job/strategy.go

```diff
@@ -113,6 +130,15 @@ func (jobStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object
 		newJob.Spec.PodFailurePolicy = nil
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.JobBackoffLimitPerIndex) {
+		if oldJob.Spec.BackoffLimitPerIndex == nil {
+			newJob.Spec.BackoffLimitPerIndex = nil
+		}
+		if oldJob.Spec.MaxFailedIndexes == nil {
+			newJob.Spec.MaxFailedIndexes = nil
+		}
```

Comment:
also need to check here that there are no new usages of the enum.

The check is simple: if FailIndex was used in the old object, you preserve all the rules. Otherwise, cleanup the rules as you did for PrepareForCreate.

Reply:
/hold 
to address this comment yet. FYI @deads2k

Reply:
@alculquicondor pod failure policy is immutable, so IIUC I cannot drop the rules on update. Now, there are two cases:
1. if the oldJob didn't contain FailIndex, the newJob also cannot, because the pod failure policy is immutable.
2. if the oldJob containerd FailIndex, then it also had to have BackoffLimitPerIndex, so the newJob also has BackoffLimitPerIndex. Because the new job has BackoffLimitPerIndex, the rules will pass validation.

So, I think it is ok, but a little convoluted.

Reply:
oh in that case just leave a comment for why we don't treat the failure policies on update.

Reply:
Done